src/distributed/redis/subservice.ts

```diff
@@ -87,11 +89,19 @@ export class RedisSubservice implements DistributedSubservice {
   }
 
   async listen(channel: string, callback: (message: any) => Promise<void>) {
-    await this.sub.subscribe(channel)
-    this.callbacks[channel] = callback
+    const scopedChannel = this.makeScopedChannel(channel)
+
+    await this.sub.subscribe(scopedChannel)
+    this.callbacks[scopedChannel] = callback
   }
 
   async send(channel: string, message: any) {
-    await this.pub.publish(channel, JSON.stringify({ nodeId: this.nodeId, ...message }))
+    const scopedChannel = this.makeScopedChannel(channel)
+
+    await this.pub.publish(scopedChannel, JSON.stringify({ nodeId: this.nodeId, ...message }))
+  }
+
+  private makeScopedChannel(key: string): string {
```

Comment:
I put the helper function at the bottom since it's private and not very useful to see before the other functions